core/src/main/java/com/linecorp/armeria/client/ClientRequestContextBuilder.java

```diff
@@ -71,9 +75,26 @@ public ClientRequestContextBuilder method(HttpMethod method) {
 
     /**
      * Sets the {@link Endpoint} of the request. If not set, it is auto-generated from the request authority.
+     *
+     * <p>Note that {@link #endpoint(Endpoint)} and {@link #endpointGroup(EndpointGroup)} are mutually exclusive.
      */
     public ClientRequestContextBuilder endpoint(Endpoint endpoint) {
-        this.endpoint = requireNonNull(endpoint, "endpoint");
+        requireNonNull(endpoint, "endpoint");
+        checkState(endpointGroup == null, "endpoint and endpointGroup are mutually exclusive.");
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the {@link EndpointGroup} of the request. If not set, an {@link Endpoint} is auto-generated from the
+     * request authority.
+     *
+     * <p>Note that {@link #endpoint(Endpoint)} and {@link #endpointGroup(EndpointGroup)} are mutually exclusive.
+     */
+    public ClientRequestContextBuilder endpointGroup(EndpointGroup endpointGroup) {
```

Comment:
This can be confusing to users because `Endpoint` is also an `EndpointGroup`. Maybe we should just remove mutual exclusiveness?

Reply:
Makes sense.